### PR TITLE
chore(server): improve worker execution speed

### PR DESCRIPTION
**Pull request by shaojunda**
1. move api and fast_jsonapi folder from lib to app/lib.  [Problems-and-Troubleshooting#autoloading](https://github.com/mperham/sidekiq/wiki/Problems-and-Troubleshooting#autoloading)
2. use find_each or each replace map.
3. reduce the number of loops inside a single job.
4. separate address consumed calculation from address balance calculation.
5. use `Set` replace `Array` when there may be duplicate content.
6. change log level to info on production.
7. save address ids in `Block` directly.
8. save node_cell_output in `CellOutput` directly.
9. add index on status to cell output.
10. add composite index on `UncleBlock` and `CkbTransaction`.

### CODE REVIEW

File: server/app/models/ckb_sync/validator.rb [hunk 1]
```diff
@@ -30,19 +30,35 @@ def call(block_hash)
...
+        local_block.ckb_transactions.find_each do |ckb_transaction|
+          cell_inputs << ckb_transaction.cell_inputs
+        end
+
+        cell_output_ids = Set.new
+        cell_inputs.flatten.each do |cell_input|
+          cell_output_ids << cell_input.previous_cell_output&.id
+        end
+
+        CellOutput.where(id: cell_output_ids.delete(nil)).update_all(status: :dead)
```
Comment (classicalliu) [inline server/app/models/ckb_sync/validator.rb]: `CellOutput.where(id: CellInput.where(ckb_transaction: lock_block.ckb_transactions).select(:id)).update_all(status: :dead)`

File: server/app/models/ckb_sync/persist.rb [hunk 2]
```diff
@@ -14,77 +14,83 @@ def sync(block_number)
...
 
-      def update_block_total_transaction_fee(block_ids)
-        Block.where(id: block_ids).map do |block|
-          block.total_transaction_fee = block.ckb_transactions.map(&:transaction_fee).reduce(0, &:+)
-
-          block
+      def block_total_transaction_fee(ckb_transactions)
+        total_transaction_fee = 0
+        ckb_transactions.each do |ckb_transaction|
+          total_transaction_fee += ckb_transaction.transaction_fee
```
[the remark this answers is not in the document]
Comment (shaojunda) [reply server/app/models/ckb_sync/persist.rb]: yes

**Comment by ashchan** [comment]
CI still fails. Please check if it's a GitHub/Travis issue or codebase problem.

**Comment by shaojunda** [comment]
I'm working on it. @ashchan , not  codebase problem

**Comment by shaojunda** [comment]
@ashchan fixed

**Comment by ashchan** [comment]
I suggest we do some benchmark to record before/after performance numbers.

**Pull request merged**